lib/diagnostic.js

```diff
@@ -17,18 +17,49 @@ const fs = require('fs');
 // resolved or rejected as appropriate
 // use the following as the new function signature
 // const sumLines = (filename) => {
-const sumLines = (filename, callback) => {
-  fs.readFile(filename, { encoding: 'utf8' }, (err, lines) => {
+const sumLines = (fileName) => {
+  const accumulator = (data) => {
     let lno = 0;
-    let sum = lines.split('\n').reduce((prev, curr, i) => {
+    let sum = data.split('\n').reduce((prev, curr, i) => {
       lno = i;
       return prev + (+curr);
     }, 0);
-    let error = isNaN(sum) && new Error(`line ${lno}: not a number`);
-    callback(error, sum);
-  });
+    if (isNaN(sum)) {
+      throw new Error(`line ${lno}: not a number`);
+    }
+
+    return sum;
+  };
+
+  const promiseReadFile = function (file, options) {
+    return new Promise((resolve, reject) => {
+      fs.readFile(file, options, (error, data) => {
+        if (error) {
+          reject(error);
+        }
+
+        resolve(data);
+      });
+    });
+  };
+
+  promiseReadFile(fileName, { encoding: 'utf8' })
```

Comment:
`sumLines()` is expected to return a promise.  This would have passed three out of the four tests if this line had been `return promiseReadFile...`;